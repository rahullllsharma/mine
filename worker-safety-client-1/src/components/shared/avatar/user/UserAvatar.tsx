import { Icon } from "@urbint/silica";
import cx from "classnames";
import { useState } from "react";
import Avatar from "../Avatar";

export default function UserAvatar({
  name,
  className,
  hoverElement,
}: {
  name?: string;
  className?: string;
  hoverElement?: JSX.Element;
}) {
  const [shouldShowHoverElement, setShouldShowHoverElement] = useState(false);

  return (
    <Avatar
      name={name || ""}
      className={cx(
        "text-xl",
        "bg-gray-200",
        "text-gray-500",
        "border",
        "border-gray-300",
        className
      )}
    >
      {hoverElement && shouldShowHoverElement ? hoverElement : null}
      <Icon
        data-testid={"user-avatar-icon"}
        onMouseEnter={() => setShouldShowHoverElement(true)}
        onMouseLeave={() => setShouldShowHoverElement(false)}
        name="user"
      />
    </Avatar>
  );
}
