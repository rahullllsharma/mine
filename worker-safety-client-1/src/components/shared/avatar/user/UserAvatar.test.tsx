import { fireEvent, render, screen } from "@testing-library/react";
import UserAvatar from "./UserAvatar";

describe("UserAvatar", () => {
  it('should render with the "UserAvatar" component', () => {
    const { asFragment } = render(<UserAvatar name="UB" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should initially not render hover component", () => {
    render(
      <UserAvatar
        hoverElement={<p data-testid="hover-test-element">Hello there</p>}
        name="UB"
      />
    );

    const hoverTestElement = screen.queryByTestId("hover-test-element");
    expect(hoverTestElement).toBeNull();
  });

  it("should render hover component on mouse enter", () => {
    render(
      <UserAvatar
        hoverElement={<p data-testid="hover-test-element">I am a user!</p>}
        name="UB"
      />
    );

    const userAvatarIcon = screen.getByTestId("user-avatar-icon");

    fireEvent.mouseEnter(userAvatarIcon);
    const hoverTestElement = screen.queryByTestId("hover-test-element");
    expect(hoverTestElement).toBeTruthy();
    expect(hoverTestElement?.textContent).toEqual("I am a user!");
  });
});
