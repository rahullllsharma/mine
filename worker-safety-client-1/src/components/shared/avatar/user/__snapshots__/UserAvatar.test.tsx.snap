// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar should render with the "UserAvatar" component 1`] = `
<DocumentFragment>
  <div
    aria-label="UB"
    class="w-7 h-7 rounded-full flex items-center justify-center text-xl bg-gray-200 text-gray-500 border border-gray-300"
    role="img"
  >
    <i
      aria-hidden="true"
      class="ci-user"
      data-testid="user-avatar-icon"
    />
  </div>
</DocumentFragment>
`;
